### ISSUE TITLE
Error messages too wordy for typed holes

### ISSUE THREAD

**Issue by codeworld-feedback** [google/codeworld#1095]
**Program:** https://code.world/#PXY-iZ1fAoeOh6sERr-CxpQ

**Comment:**

The error message for holes is incredibly long and wordy.  That's too bad, because I'm convinced that holes are a good strategy for getting students started with stubbing out parts of programs.

**Message:**

    program.hs:47:17: warning:
        • Found hole: _ :: [Truth] -> Truth
        • In the expression: _
          In the expression:
            _ ([collidesObstacle (ox, oy, x, y) |
                  (ox, _) <- first (obs, 10) |  oy <- [0, 4 .. ]])
          In an equation for lost:
              lost (Playing (obs, (x, y), _, h, score, bar))
                | x < - 9 = True
                | x > 9 = True
                | y < h - 9 = True
                | y < bar + 1.1 = True
                | otherwise
                = _ ([collidesObstacle (ox, oy, x, y) |
                        (ox, _) <- first (obs, 10) |  oy <- [0, 4 .. ]])
        • Relevant bindings include
            bar :: Number (bound at program.hs:42:39)
            score :: Number (bound at program.hs:42:32)
            h :: Number (bound at program.hs:42:29)
            y :: Number (bound at program.hs:42:22)
            x :: Number (bound at program.hs:42:20)
            obs :: [(Number, Color)] (bound at program.hs:42:14)
            (Some bindings suppressed; use -fmax-relevant-binds=N or -fno-max-relevant-binds)
          Valid hole fits include
            all :: [Truth] -> Truth
              (defined in the standard library
               (and originally defined in codeworld-
    Truth)
              (defined in the standard library
               (and originally defined in the standard library))

**Comment by cdsmith** [comment]
This work was finished in #1097
